spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/IntegrationFlowFunctionSupport.java

```diff
@@ -211,4 +200,8 @@ public IntegrationFlowBuilder integrationFlowFromProvidedSupplier(Supplier<?> su
 		subscribeToOutput(outputProcessor, functionInvoker.apply((Flux<Message<I>>) inputPublisher)).subscribe();
 	}
 
+	@Override
+	public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.errorChannel = (MessageChannel) applicationContext.getBean(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME);
```

Comment:
Would be better to move this resolution to the `afterPropertiesSet()` if we can't inject an `errorChannel` from the `@Configuration` level.